### ISSUE TITLE
Reopen: Missing Recipes Solved

### ISSUE THREAD

**Issue by GracefulUni** [blaxxun-boop/Backpacks#54]
Hi Blaxxun,
Based on my previous issue that you closed, I finally figured out the cause of the missing recipes disappearing compared to the rest of my backpacks that are not gone.
Half of my backpacks do NOT have an upgrade recipe, while the ones in image do have a upgrade recipe, I added upgrade recipes to one of the backpacks and the bug happened where all my other recipes disappeared, but the one that I recently added an upgrade recipe to it that one didnt disappear from the crafting list.

TLDR, if the backpacks do not have an upgrade recipe there is a chance that the recipe will disappear from the crafting bench, I tested this twice already.

**Comment by BETLOG** [comment]
I just compiled a bunch of packs with zero upgradeability, and had no issues.

WIP, but seems to work ok: 
[Backpacks_betlog.zip](https://github.com/user-attachments/files/17613490/Backpacks_betlog.zip)

FYI:
I did notice that things can get weird and you might need to re-aquire ingredients before the recipe appears. (as mentioned in another issue here)
I have also started destroying  all prior versions that exist as ingame objects (get an obliterator), partly to reduce confusion, and partly to make sure there isnt a cache somewhere tripping over itself.
Dont put comments at end of yaml lines. !!!
Use the configurationmanager yaml editor that is available when you toggle custom yaml on and a valid file is detected, just to check if the syntax is ok. You can still edit the yaml externally and it autoupdates when you save.

**Comment by blaxxun-boop** [comment]
@BETLOG Please note that "no license" on GitHub means, you aren't allowed to touch the code _at all_.

**Comment by BETLOG** [comment]
> @BETLOG Please note that "no license" on GitHub means, you aren't allowed to touch the code _at all_.

I'm looking for the backpacks licence to use as a guide, but i cant see any...
is a line in the yaml saying `# LICENSE : public domain` enough? I assume thats the code you refer to.

**Comment by BETLOG** [comment]
lol, after saying it works fine and implying more care needed, AND after experiencing this numerous months ago last time I used backpacks (so i should know better already)... i just went to poke around in my WIP backpacks that were working fine for hours while i broke and fixed them repeatedly last night, which haven't been changed since then except to add `# LICENSE : public domain` to the first line....and now they were completely absent from the workbench.
What ended up working is to reset (set `off`) `external yaml` in configurationmanager, and then re-enable it, check my yaml was detected, loaded and syntax valid, and hey presto the backpacks list on the workbench again.
Yeah, this is a longtime recurring bug thats probably going to be hard to duplicate.
